### DISCUSSION TITLE
Does Mend-hosted Github app have an execution timeout?

### DISCUSSION THREAD

**Discussion by ogoshikazuki** [renovatebot/renovate#26589] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

The job status frequently becomes FAILED. Comparing the log with a job whose status is DONE, it appears that the job has not been executed to the end.

Last in the log of JOBs whose status is DONE:
```
INFO: Repository finished
{
  "cloned": false
  "durationMs": 1763244
}

DEBUG: Renovate exiting
```

(e.g.) The end of the log for a JOB with a status of FAILED:
```
DEBUG: exec completed (branch="renovate/group-application-console-f")
{
  "durationMs": 2338
  "stdout": "[08:53:14.804] INFO (328): Installing npm pnpm@8.12.0...\n8.12.0\n[08:53:16.206] INFO (328): Installed tool pnpm in 1.4s.\n"
  "stderr": ""
}

DEBUG: Executing command (branch="renovate/group-application-console-f")
{
  "command": "pnpm install --recursive --lockfile-only --ignore-scripts --ignore-pnpmfile"
}
```

It appears that jobs with a status of FAILED are generally forced out after 30 minutes.
Does Mend-hosted Github app have an execution timeout?
If there is an execution timeout, is there any workaround?

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: exec completed (branch="renovate/group-application-console-f")
{
  "durationMs": 2338
  "stdout": "[08:53:14.804] INFO (328): Installing npm pnpm@8.12.0...\n8.12.0\n[08:53:16.206] INFO (328): Installed tool pnpm in 1.4s.\n"
  "stderr": ""
}

DEBUG: Executing command (branch="renovate/group-application-console-f")
{
  "command": "pnpm install --recursive --lockfile-only --ignore-scripts --ignore-pnpmfile"
}

```

</details>

**Comment by ogoshikazuki** [comment]
What do you think of the above?

**Comment by rarkins** [comment, marked as answer]
Yes there is a timeout,
You can self host with the GitHub Action if you need jobs lastomg longer than 30 minutes
